File: examples/sample_module/calc.py
"""
Модуль калькулятора для демонстрации связи код ↔ документация.
"""


def add(a: int, b: int) -> int:
    """Складывает два числа."""
    return a + b


def subtract(a: int, b: int) -> int:
    """Вычитает b из a."""
    return a - b


class Calculator:
    """Простой калькулятор."""

    def multiply(self, x: float, y: float) -> float:
        """Умножает x на y."""
        return x * y

    def divide(self, a: float, b: float) -> float:
        """Делит a на b. При b=0 бросает ZeroDivisionError."""
        return a / b

File: examples/go_sample/calc.go
// Package calc provides calculator functions for LiveDoc demo.
package calc

// Add returns the sum of a and b.
func Add(a, b int) int {
	return a + b
}

// Subtract returns a minus b.
func Subtract(a, b int) int {
	return a - b
}

// Calculator holds state for chained operations.
type Calculator struct {
	value int
}

// Multiply multiplies x and y.
func (c *Calculator) Multiply(x, y int) int {
	c.value = x * y
	return c.value
}

// Divide returns a divided by b.
func (c Calculator) Divide(a, b int) int {
	return a / b
}
